app/controllers/api/discussions_controller.rb

```diff
@@ -55,37 +49,19 @@ def discussion_params
   end
 
   def visible_records
-    load_and_authorize_group
-    groups = if @group
-               VisibleGroupsQuery.expand(group: @group, user: current_user)
-             else
-               current_user.groups
-             end
-    Queries::VisibleDiscussions.new(user: current_user, groups: groups).order(last_activity_at: :desc)
+    Queries::VisibleDiscussions.new(user: current_user, groups: current_user.groups)
   end
 
   private
-  def filter_discussions(discussions, filter)
-    case filter
-    when 'show_unread'    then discussions.unread
-    when 'show_proposals' then discussions.with_active_motions
-    else                       discussions
-    end
-  end
-
-  def inbox_threads
-    discussions = Queries::VisibleDiscussions.new(groups: current_user.groups,
-                                                  user: current_user) 
-    discussions = filter_discussions(discussions, params[:filter])
-    discussions.not_muted.
-                where('last_activity_at > ?', params[:from_date] || 3.months.ago).
-                joined_to_current_motion.
-                preload(:current_motion, {group: :parent}).
-                order('motions.closing_at ASC, last_activity_at DESC')
-  end
 
-  def grouped(discussions)
-    discussions.map { |g, discussions| discussions.first(Integer(params[:per] || 5)) }.flatten
+  def discussions_for_preview(filter = params[:filter])
+    case filter
+    when 'show_proposals'     then visible_records.not_muted.with_active_motions
+    when 'show_participating' then visible_records.not_muted.participating
+    when 'show_muted'         then visible_records.muted
+    when 'show_unread'        then visible_records.not_muted.unread
+    else                           visible_records.not_muted
+    end.recent.sorted_by_latest_motions
```

Comment:
This looks really great. We must know what we're doing..

I'm just confused as to why there is still sorted_by_latest_motions on there..